summary/README.md

```diff
@@ -0,0 +1,758 @@
+
+### Empirical Study
+This empirical study aims at answering the following three questions:
+- **Q1:** Are there coverage goals for which the traditional *OneBranch* approach performs better?
+- **Q2:** How many coverage goals found by the *Whole* approach get missed by *OneBranch*?
+- **Q3:** Which factors influence the relative performance between *Whole* and *OneBranch*?
+
+100 Java classes were randomly chosen out of 100 randomly selected projects. 
+
+#### Q1
+After performing the empirical study, it was found that *OneBranch* performed better for over 50 coverage goals. Three of these coverage goals were never covered by *Whole*. However, there are more than half cases for which *Whole* performed much better. In other words, even with three difficult goals that *OneBranch* can cover, many more (82 times) difficult branches were covered by *Whole* and ignored by *OneBranch*.
+
+#### Q2
+As mentioned before, *Whole* is able to handle 82 times more difficult branches than *OneBranch*. It is then assessed that the *Whole* approach leads to higher coverage, even for difficult branches. 
+
+#### Q3
+The researchers quantified the odds that *Whole* performs better to cover a certain goal compared to *OneBranch*. After some research and high usage of statistical techniques, there was no correlation that could be found between quantification of performance and other factors. This means that there is no evidence of a factor that strongly influences the relative performance between *Whole* and *OneBranch*. 
+
+### Conclusion
+Existing research has shown that the whole test suite approach lead to better coverage results than the traditional one target at a time approach. There was reasonable doubt on whether this would be the case when targetting difficult goals. This paper performed an in-depth analysis to study if this doubt could be confirmed in practice. It was found that indeed, the *Whole* approach had more difficulties when targetting difficult goals. However, these cases are very few compared to the cases for which *Whole* performed significantly better. This paper therefore provides more support to the validity and usefulness of the whole test suite generation method in the context of software testing.
+
+
+## Reformulating Branch Coverage As A Many-Objective Optimization Problem [[6](#6)]
+
+### Test Data Generation
+Represents a search problem that aims at testing as many program element as possible (to maximize the number of branches).
+Whole test suite approach that was previously described as preferable to the single-branch at a time method.
+A new branch coverage will be proposed as many-objective optimisation problem alternatively to whole test suite approach that aggregates multiple objectives into a single value.
+A highly scalable many-objective genetic algorithm, called MOSA(Many-Objective Sorting Algorithm) is proposed to many-objective branch coverage problem.
```

Comment:
`, called MOSA(Many-Objective Sorting Algorithm) is proposed to many-objective...` -> `, Many-Objective Sorting Algorithm (MOSA), is proposed to relieve the many...`